evm/chains/chain.py

```diff
@@ -1,405 +1,7 @@
-from __future__ import absolute_import
-import logging
-
-from cytoolz import (
-    assoc,
-)
-
-from eth_utils import (
-    to_tuple,
-)
-
-from evm.consensus.pow import (
-    check_pow,
-)
-from evm.constants import (
-    MAX_UNCLE_DEPTH,
-)
-from evm.estimators import (
-    get_gas_estimator,
-)
-from evm.exceptions import (
-    BlockNotFound,
-    TransactionNotFound,
-    ValidationError,
-    VMNotFound,
-)
-from evm.validation import (
-    validate_block_number,
-    validate_uint256,
-    validate_word,
+from .base import (
+    BaseChain,
 )
-from evm.rlp.headers import (
-    BlockHeader,
-)
-from evm.utils.chain import (
-    generate_vms_by_range,
-)
-from evm.utils.datatypes import (
-    Configurable,
-)
-from evm.utils.headers import (
-    compute_gas_limit_bounds,
-)
-from evm.utils.hexadecimal import (
-    encode_hex,
-)
-from evm.utils.rlp import (
-    ensure_imported_block_unchanged,
-)
-
-
-class Chain(Configurable):
```

Comment:
Note to self: make sure that the latest and `Chain` is moved to `BaseChain` completely.